nautobot/core/management/commands/generate_performance_test_endpoints.py

```diff
@@ -0,0 +1,269 @@
+    """
+
+    help = "List all relevant performance test url patterns in Nautobot Core"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--output-file",
+            default=False,
+            help="Flag to whether load the output the endpoints to a `./endpoints.yaml` file",
+        )
+
+    def handle(self, *args, **options):
+        # Get the URL resolver
+        url_patterns = get_resolver().url_patterns
+
+        # Group the urls by app names
+        self.app_name_to_urls = {}
+        self.app_name_to_urls["endpoints"] = {}
+        # Fetch and store the urls by app names in the dictionary
+        self.fetch_urls(url_patterns)
+        for view_name, url_patterns in self.app_name_to_urls["endpoints"].items():
+            # De-duplicate the URL patterns and sort them.
+            self.app_name_to_urls["endpoints"][view_name] = sorted(list(set(url_patterns)))
+
+        self.stdout.write(yaml.dump(self.app_name_to_urls, sort_keys=True))
+
+        if options.get("output_file"):
+            with open("./endpoints.yml", "w") as outfile:
+                yaml.dump(self.app_name_to_urls, outfile, sort_keys=True)
+
+    def is_eligible_get_endpoint(self, view_name):
+        """
+        Check if the view is a GET endpoint and if it is eligible for performance testing.
+        """
+        if view_name not in EXCLUDED_VIEW_NAMES and (view_name.endswith(GET_ENDPOINT_SUFFIXES) or "_" not in view_name):
+            return True
+        return False
+
+    def append_urls_to_dict(self, url_pattern, model_class, view_name, is_api_endpoint=False):
+        """
+        URL patterns are stored in the dictionary in the following format:
+            - Any model detail view URL pattern that contains `<uuid:pk>` or `(?P<pk>[/.]+)` will have two endpoints:
+                - One with the `model_class.objects.first().pk`
+                - One with the `model_class.objects.last().pk`
+            - Any model list view URL pattern will have two endpoints:
+                - One with default pagination
+                - One with custom pagination (5 pages with <total_object_count//5> instances per page)
+            - Any generic endpoint like `core:home` will have one endpoint which is the URL pattern itself.
+        """
+        if not model_class:
+            # A generic endpoint like `core:home`
+            if view_name not in self.app_name_to_urls["endpoints"]:
+                self.app_name_to_urls["endpoints"][view_name] = []
+            self.app_name_to_urls["endpoints"][view_name].append(url_pattern)
+            return
+
+        # If the model class is found, then we know we are dealing with a model related endpoint
+        if len(model_class.objects.all()) < 2:
+            # TODO handle the case where there is no instances of the model is found
+            self.stderr.write(f"Not enough instances of {model_class} found, need at least 2")
+            return
```

Comment:
If there's exactly one instance, we can still at least provide a `first_url_pattern`, can't we?

Actually, even if there are zero instances, the default list URL would still be good to document/test, wouldn't it?

Reply:
A little push-back on the zero instances case, as I think it will already be covered by "-list" and "_list" endpoints.
But definitely we can add another case for exactly one instance.

Reply:
Ah okay, I may have misread the code, I thought this early return was preventing us from evaluating the list endpoints.

Reply:
Can you confirm?

Reply:
Sorry you are right, it did return early. Gonna push up a fix now